instructions/rate-limiting.adoc

```diff
@@ -1,98 +1,203 @@
+customer   [customer-gateway]   [*]     18h
+----
+====
+
+[#ratelimiting]
+== Rate Limiting
+
+IMPORTANT: The Rate Limiting rules take some time to be applied and reflected. Be patient here!
+
+Here we will limit the number of concurrent requests into recommendation v2
+
+Now apply the rate limiting. To see how this is done,
+take a look at link:http://github.com/thoraxe/istio-lab-summit-2019/blob/master/src/istiofiles/recommendation_rate_limit_handler.yml[recommendation_rate_limit_handler.yml]
+
+[source,bash,subs="+macros,+attributes"]
+----
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: recommendation
+  namespace: istio-tutorial
+spec:
+  host: recommendation
+  subsets:
+  - labels:
+      version: v1
+    name: v1
+  - labels:
+      version: v2
+    name: v2
+  - labels:
+      version: v3
+    name: v3
+---
+apiVersion: config.istio.io/v1alpha2
+kind: memquota
+metadata:
+  name: handler
+  namespace: istio-system
+spec:
+  quotas:
+  - name: requestcount.quota.istio-system
+    maxAmount: 500
+    validDuration: 1s
+    # The first matching override is applied.
+    # A requestcount instance is checked against override dimensions.
+    overrides:
+    # The following override applies to 'reviews' regardless
+    # of the source.
+    - dimensions:
+        destination: recommendation
+      maxAmount: 4
+      validDuration: 5s
+---
+apiVersion: config.istio.io/v1alpha2
+kind: quota
+metadata:
+  name: requestcount
+  namespace: istio-system
+spec:
+  dimensions:
+    source: request.headers["x-forwarded-for"] | "unknown"
+    destination: destination.labels["app"] | destination.service.name | "unknown"
+    destinationVersion: destination.labels["version"] | "unknown"
+---
+apiVersion: config.istio.io/v1alpha2
+kind: QuotaSpec
+metadata:
+  name: request-count
+  namespace: istio-system
+spec:
+  rules:
+  - quotas:
+    - charge: 1
+      quota: requestcount
+---
+apiVersion: config.istio.io/v1alpha2
+kind: QuotaSpecBinding
+metadata:
+  name: request-count
+  namespace: istio-system
+spec:
+  quotaSpecs:
+  - name: request-count
+    namespace: istio-system
+  services:
+  - name: recommendation
+    namespace: istio-tutorial
+    #  - service: '*'  # Uncomment this to bind *all* services to request-count
+---
+apiVersion: config.istio.io/v1alpha2
+kind: rule
+metadata:
+  name: quota
+  namespace: istio-system
+spec:
+  # quota only applies if you are not logged in.
+  # match: match(request.headers["cookie"], "user=*") == false
+  actions:
+  - handler: handler.memquota
+    instances:
+    - requestcount.quota
+----
+
+Note that the memquota resource `handler` defaines a rate limit of 4 requests per 5 seconds for the `recommendation`
+destination. Next, deploy the rate limiting resources.
+
+[source,bash,subs="+macros,+attributes"]
+----
+$ oc -n istio-tutorial create -f http://github.com/thoraxe/istio-lab-summit-2019/blob/master/src/istiofiles/recommendation_rate_limit.yml
```

Comment:
oc -n istio-tutorial create -f src/istiofiles/recommendation_rate_limit.yml
destinationrule.networking.istio.io/recommendation created
the namespace from the provided object "istio-system" does not match the namespace "istio-tutorial". You must pass '--namespace=istio-system' to perform this operation.
the namespace from the provided object "istio-system" does not match the namespace "istio-tutorial". You must pass '--namespace=istio-system' to perform this operation.
the namespace from the provided object "istio-system" does not match the namespace "istio-tutorial". You must pass '--namespace=istio-system' to perform this operation.
the namespace from the provided object "istio-system" does not match the namespace "istio-tutorial". You must pass '--namespace=istio-system' to perform this operation.
the namespace from the provided object "istio-system" does not match the namespace "istio-tutorial". You must pass '--namespace=istio-system' to perform this operation.

Reply:
The yaml file appears to be installing into two namespaces so try and run it without the '-n' option